app/controllers/api/v1/accounts/relationships_controller.rb

```diff
@@ -7,7 +7,10 @@ class Api::V1::Accounts::RelationshipsController < Api::BaseController
   respond_to :json
 
   def index
-    @accounts = Account.where(id: account_ids).select('id')
+    accounts = Account.where(id: account_ids).select('id')
+    # .where doesn't guarantee that our results are in the same order
+    # we requested them, so return the "right" order to the requestor.
+    @accounts = accounts.index_by(&:id).values_at(*account_ids)
```

Comment:
We should be ordering in SQL as much as possible.

Reply:
In general, I agree. However, the API for this function allows users to supply an arbitrarily-ordered set of IDs, and the tests imply they're expected to be returned in the same order (although I can't find any docs saying one way or the other). Doing this ordering in SQL would require synthesizing a CASE to sort by, which seems unlikely to be cleaner or faster.

In practice, this should be roughly O(n) (to the extent that you believe hashes are really O(1) insertion/lookup), with very small values of n. (Also in practice, the endpoint currently doesn't actually guarantee you get entries back in the same order you requested them, so we could just toss this change.)

If you think we should generate the CASE and do it in SQL, I can write that code, it just felt unnecessary here.

Reply:
My understanding is that our code doesn't depend on this anywhere, and it was just ease of use for the tests. I can ask app developers though.